### PR TITLE
Add missing tests

### PR DESCRIPTION
**Pull request by hjdivad**
This adds a couple of failing tests and a passing but previously missing test.

The missing test added is for calling `set` which was destructured from `import Ember from 'ember'`

One failing test is to handle the case where `Ember.computed` is called from an import that isn't named `Ember`, eg
```
import E from 'ember';
E.computed()
```

Another failing test illustrates that ember import bindings are clobbered if followed by any other default import, as in:
```
import Ember from 'ember';
import AnythingElse from 'anywhere';
```

tl;dr things to fix
- [ ] don't clobber ember import bindings
- [ ] move module scope information to `context` arg
- [ ] more intelligently detect `Ember.computed`

### CODE REVIEW

**Comment by hjdivad** [comment]
cc @rwjblue

**Comment by rwjblue** [comment]
Hehe. Confirm. Really need to get that wired up right through testem and whatnot.

**Comment by rwjblue** [comment]
OK, just did a pass across this rule to fix the largest issues that @hjdivad pointed out above.  I pushed the fix commits directly to this branch, so it should be ready for review...

File: tests/lib/rules/no-side-effect-cp.js [hunk 1]
```diff
@@ -228,6 +242,55 @@ ruleTester.run('no-side-efffect-cp', rule, {
...
+    },
+    // This test must come before any others that `import Ember from 'ember'`
+    // to illustrate the problem of the import binding being saved in module
+    // scope.
+    // see #91
+    {
+      code: `
+      import Ember from 'ember';
+      import SomethingElse from 'something-else';
+      const { set } = Ember;
```
Comment (scalvert) [inline tests/lib/rules/no-side-effect-cp.js]: Can we tab these lines in to line them up with the code below. It's a bit hard to read like this.

File: tests/lib/rules/no-side-effect-cp.js [hunk 2]
```diff
@@ -228,6 +242,55 @@ ruleTester.run('no-side-efffect-cp', rule, {
...
+          })
+        });`,
+      errors: [{
+        message: MESSAGE
+      }]
+    },
+    {
+      code: `
+      import Ember from 'ember';
+      const { set } = Ember;
```
Comment (scalvert) [inline tests/lib/rules/no-side-effect-cp.js]: Same here re: spacing.

File: tests/lib/rules/no-side-effect-cp.js [hunk 3]
```diff
@@ -228,6 +242,55 @@ ruleTester.run('no-side-efffect-cp', rule, {
...
+            set('baz', 'wat');
+          })
+        });`,
+      errors: [{
+        message: MESSAGE
+      }]
+    },
+    {
+      code: `
+      import E from 'ember';
```
Comment (scalvert) [inline tests/lib/rules/no-side-effect-cp.js]: Same here re: spacing.

File: lib/utils/computed-property.js [hunk 4]
```diff
@@ -51,18 +51,20 @@ function isCPAccessor(node) {
...
     let caller = getCaller(callee);
-    return caller === 'Ember.computed' || caller === 'Em.computed';
+    return caller === 'Ember.computed' || caller === 'Em.computed' || caller === importedEmberName + '.computed';
   }
   return false;
 }
 
-module.exports = function isCPGetter(node) {
-  return isCPDesc(node, 'get') || isCPAccessor(node) || isPrototypeExtCP(node);
+module.exports = function isCPGetter(node, importedCPName, importedEmberName) {
```
Comment (scalvert) [inline lib/utils/computed-property.js]: For some reason it feels more natural to have `importedEmberName` come  before `importedCPName`, like in hierarchical order.

File: lib/utils/computed-property.js [hunk 5]
```diff
@@ -22,12 +22,12 @@ function isCPDesc(node, method) {
...
   }
   let callee = greatGrandParent.callee;
-  if (greatGrandParent.type === 'Identifier' && callee.name === 'computed') {
+  if (greatGrandParent.type === 'Identifier' && (callee.name === 'computed' || callee.name === importedCPName)) {
     return true;
   }
   if (callee.type === 'MemberExpression') {
     let caller = getCaller(callee);
-    return caller === 'Ember.computed' || caller === 'Em.computed';
+    return caller === 'Ember.computed' || caller === 'Em.computed' || caller === importedEmberName + '.computed';
```
Comment (scalvert) [inline lib/utils/computed-property.js]: You could template string `\`${importedEmberName}.computed\``

**Comment by hjdivad** [comment]
LGTM 👍

**Comment by rwjblue** [comment]
@scalvert - I believe that the latest push addresses all of your comments.  Let me know if you spot anything else...

**Comment by scalvert** [comment]
LGTM
